base/io.jl

```diff
@@ -884,9 +884,9 @@ end
 
 function read(io::IO, ::Type{Char})
     b0 = read(io, UInt8)::UInt8
-    l = 8(4-leading_ones(b0))
+    l = 0x08 * (0x04 - UInt8(leading_ones(b0)))
     c = UInt32(b0) << 24
-    if l < 24
+    if l ≤ 0x10
```

Comment:
```suggestion
    if l < 0x18
```
if we want to make it the same as before.  (LLVM seems to generate essentially the same code, though?)

Reply:
Eh, 🤷🏻‍♂️